models/repo.go

```diff
@@ -1387,6 +1387,36 @@ func DeleteRepository(uid, repoID int64) error {
 		RemoveAllWithNotice("Delete attachment", attachmentPaths[i])
 	}
 
+	// Remove LFS objects
+	var lfsObjects []*LFSMetaObject
+
+	if err = sess.Where("repository_id=?", repoID).Find(&lfsObjects); err != nil {
+		return err
+	}
+
+	for _, v := range lfsObjects {
+
+		//Check that no other repository references this object
+		var matchedOidObjects []*LFSMetaObject
+		if err = sess.Where("oid=?", v.Oid).Find(&matchedOidObjects); err != nil {
+			return err
+		}
+
+		if len(matchedOidObjects) > 1 {
+			continue
+		}
+
+		oidPath := filepath.Join(v.Oid[0:2], v.Oid[2:4], v.Oid[4:len(v.Oid)])
+		err = os.Remove(filepath.Join(setting.LFS.ContentPath, oidPath))
+		if err != nil {
+			return err
+		}
+	}
+
+	if _, err = sess.Exec("DELETE FROM `lfs_meta_object` WHERE repository_id=?", repoID); err != nil {
```

Comment:
Changed in https://github.com/go-gitea/gitea/pull/122/commits/a2945dd431edc448f94cadc72c7168bbb2e4cc03